main.cpp

```diff
@@ -13,75 +13,424 @@ What you create in this project will be used as the basis of Project 5 in the co
+    5) height in cm (int)
+3 things it can do:
+    1) display text
+    2) regulate brightness
+    3) connect to the motherboard
+ */
+
+struct Display
+{
+    int numberPixels = 1800;
+    float powerConsumed = 180.f;
+    double bightness = 70;
+    int width = 250;
+    int height = 100;
+
+    void displayText(std::string text);
+    bool regulatBrightness(int brightnessLevel);
+    bool connectToMotherBoard(std::string connectorType = "HDMI");
+};
+
+/*
+Thing 7) Bettery
+5 properties:
+    1) voltage (float)
+    2) discharge time (double)
+    3) capacity (double)
+    4) power density (double)
+    5) surface area in mm^2 (float)
+3 things it can do:
+    1) charge
+    2) discharge
+    3) monitor it's capacity
+ */
+
+ struct Battery 
+ {
+    float voltage = 4.7f;
+    double dischargeTime = 5;
+    double capacity = 150;
+    double powerDensity = 4870;
+    float surfaceArea = 150.75;
+
+    void charge(int time);
+    void discharge(int time);
+    void monitorCapacity(bool inforUser = false);
+ };
+
+/*
+Thing 8) Operating system
+5 properties:
+    1) number of multitaking operation (long)
+    2) build number (str::string)
+    3) version (str::string)
+    4) name (str::string)
+    5) provider (str::string)
+3 things it can do:
+    1) run application
+    2) install application
+    3) uninstall application
+ */
+
+ struct OperatingSystem
+ {
+    long numberOfConcurrentTask = 15785;
+    std::string buildNumnber = "654f6";
+    std::string version = "Apha 4";
+    std::string name = "Windows";
+    std::string provider = "Micro$oft";
+
+    void runApplication(std::string applicationName);
+    bool installApplication(std::string appName);
+    bool uninstallApplication(std::string appName);
+ };
+
+/*
+Thing 9) Memory
+5 properties:
+    1) capacity (int)
+    2) access method (str::string)
+    3) unit of transfer (str::string)
+    4) lenght (int)
+    5) performance (int)
+3 things it can do:
+    1) load data
+    2) offload data
+    3) access memory block
+ */
+
+struct Memory
+{
+    int capacity = 1524556;
+    std::string accessMethod = "random access";
+    std::string unitOfTransfer = "256 bits";
+    int lenght = 100;
+    int performance = 150;
+
+    void loadData(int amountOfData);
+    void offloadData(int amountOfData, bool writeToSWAP = false);
+    int accessMemoryBlock(int blockLocation);
+};
+
+/*
+Thing 10) Computer
+5 properties:
+    1) Keyboard
+    2) Display
+    3) Bettery
+    4) Operating system
+    5) Memory
+3 things it can do:
+    1) compile C++ code
+    2) run productivity applications
+    3) connect peripherals
+ */
+
+struct Computer
+{
+    Keyboard keyboard;
+    Display display;
+    Battery battery;
+    OperatingSystem os;
+    Memory memory;
+
+    bool compileCode(Memory memory, std::string codeLanguage = "c++");
+    bool runProductivityApp(OperatingSystem supportedOperatingSystem, std::string applicationName);
+    bool connectPeripherals(Keyboard keyboard);
```

Comment:
This is a common issue with new programmers or programmers who are coming from garbage-collected languages (python, C#).
When you pass this type into this function, the instance that is used inside this function is NOT the same instance that was passed in.
That means any changes you make to the instance being used INSIDE the function are NOT kept when this function ends.
the C++ concept of References and Pointers has not been covered at this point in the course. If you wanted this to work as you are most likely expecting it to work, you would need to use references or pointers.
I don't let people use coding concepts in projects until after they've been covered in the course.
You don't need to revise these function parameters.
I just want you to be aware of the potential issue.
- [ ] acknowledge you understand this.
- [ ] tell me in slack if you decide to revise or wish to leave the code as is.